pkg/trace/writer/trace.go

```diff
@@ -49,32 +50,39 @@ type TraceWriter struct {
 	// Channel should only be received from when testing.
 	In chan *SampledSpans
 
-	hostname string
-	env      string
-	senders  []*sender
-	stop     chan struct{}
-	stats    *info.TraceWriterInfo
-	wg       sync.WaitGroup // waits for gzippers
-	tick     time.Duration  // flush frequency
+	hostname  string
+	env       string
+	senders   []*sender
+	stop      chan struct{}
+	flushChan chan chan struct{}
+	stats     *info.TraceWriterInfo
+	wg        sync.WaitGroup // waits for gzippers
+	tick      time.Duration  // flush frequency
 
 	traces       []*pb.APITrace // traces buffered
 	events       []*pb.Span     // events buffered
 	bufferedSize int            // estimated buffer size
+	syncMode     bool           // when enabled, traces are only sent when Flush is called
 
 	easylog *logutil.ThrottledLogger
 }
 
 // NewTraceWriter returns a new TraceWriter. It is created for the given agent configuration and
 // will accept incoming spans via the in channel.
 func NewTraceWriter(cfg *config.AgentConfig) *TraceWriter {
+
+	in := make(chan *SampledSpans, 1000)
+
```

Comment:
Why did you move this out? Can we put it back? I don't see the var used anywhere.